### PR TITLE
[calyx-py] Add case statement support for ComponentBuilders

### PR DESCRIPTION
**Pull request by nathanielnrn**
Implements #2166.

This goes with the "control-style" implementation mentioned there. And it can be noted that this wraps all of the `if`s generated into a single `par` block.

### CODE REVIEW

**Comment by anshumanmohan** [comment]
Thanks for the fix! Happy for you to merge

**Review by anshumanmohan — APPROVED** [review]

File: calyx-py/calyx/py_ast.py [hunk 1]
```diff
@@ -362,6 +376,10 @@ class Atom(GuardExpr):
     def doc(self) -> str:
         return self.item.doc()
 
+    @property
+    def name(self) -> str:
+        return f"{self.item.get_name()}"
+
```
Comment (nathanielnrn) [inline calyx-py/calyx/py_ast.py:382]: @anshumanmohan [not urgent] Requested a re-review to ask if you could glance at this `get_name` stuff when you get a chance? It was surprisingly annoying to get right.

Comment (anshumanmohan) [reply calyx-py/calyx/py_ast.py:382]: Aw man yeah it does look rough. Thanks for doing it, the case statements do look much more readable at least to my eye!

**Review by anshumanmohan — CHANGES_REQUESTED** [review]
I took a look at your name-extraction thing and it does look like pulling teeth. Thanks for getting it right!!

Edit: ignore the rest, it was a silly mistake on my end.
~Unrelated, and perhaps more important: this investigation prompted me to look at my `fifo.futil` code a little more closely, and I noticed an unnecessary double-par wrapping. I've checked the file in; it was generated with `python3 calyx-py/test/correctness/queues/fifo.py 20000 > calyx-py/test/correctness/queues/fifo.futil`.~

File: calyx-py/test/correctness/queues/fifo.futil [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+    cmd_eq_1.left = cmd;
+    cmd_eq_1.right = 2'd1;
+    cmd_eq_2.left = cmd;
+    cmd_eq_2.right = 2'd2;
+    cmd_eq_3.left = cmd;
+    cmd_eq_3.right = 2'd3;
+  }
+  control {
+    par {
+      par {
```
Comment (anshumanmohan) [inline calyx-py/test/correctness/queues/fifo.futil:102]: This control was generated using a `case` helper, and it double-wrapped in `par`.

**Comment by anshumanmohan** [comment]
Oh never mind, that double-par was totally on my end. Fixed, and I will remove the "request changes" thing now.

**Review by anshumanmohan — APPROVED** [review]

**Comment by nathanielnrn** [comment]
Thanks much! Will merge when tests pass

**Pull request merged**